### DISCUSSION TITLE
tutorials/voyage-part1/02-project-setup

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#65] [course feedback]
# tutorials/voyage-part1/02-project-setup

Set up the starter code for FlyBy, the demo app you'll be building in this course.

https://www.apollographql.com/tutorials/voyage-part1/02-project-setup

**Comment by mlindes** [comment]
The project code is out of date..  My npm is up to date at v20.5.0 but the project requries version `<=18`. Would be great if this project could be brought up to date. 

```
~/git/apollographql/odyssey-voyage-I/subgraph-locations (main 🟢) »npm install  
npm ERR! code EBADENGINE
npm ERR! engine Unsupported engine
npm ERR! engine Not compatible with your version of node/npm: location-reviews@1.0.0
npm ERR! notsup Not compatible with your version of node/npm: location-reviews@1.0.0
npm ERR! notsup Required: {"npm":">=6.0.0","node":">=14.0.0 <=18"}
npm ERR! notsup Actual:   {"npm":"9.8.0","node":"v20.5.0"}

npm ERR! A complete log of this run can be found in: /Users/mlinde201/.npm/_logs/2023-09-01T17_41_34_900Z-debug-0.log
```

**Reply by mabuyo** [reply]
Hey there @mlindes! I'll put this on the list to take a look at. In the meantime, you can try using node v18, or you can remove the `engines` property in `package.json` to run the project without needing to meet those node requirements.

Let us know if you run into any issues.

**Reply by yogeshminocha** [reply]
Did a slight update to package.json, and it worked. 

 "node": ">=14.0.0 <=22"

**Reply by tsinghammer** [reply]
I'm using nvm since I have to juggle between different Node version anyway. But if it's fully compatible, it would be great to update the package.json as @yogeshminocha suggested. Are contributions welcome? I could to a TypeScript version as suggested below.

**Comment by laeeqFaiz** [comment]
There is no typescript in Voyage... :(

**Reply by Dark417** [reply]
I learned more typescript in the previous series than GQL.. would be great if that continues

**Reply by lizhennessy** [reply]
Hey there @ignicubeLaeeq and @Dark417 ! Thanks so much for your feedback. Bringing a TypeScript option into the Voyage series is most definitely on our roadmap! We hope to supplement our courses with this feature early in the new year. Please let us know if there are any other language or features you'd find helpful on your Odyssey journey!

**Comment by torressam333** [comment]
Is the Serverless For developers building a GraphQL API. Try out GraphOS without any additional ops work. $0 up to 10M requests / month Plan enough for us to complete the course in it's entirety? Or do we need the full enterprise plan for this course and series? Thanks!

**Reply by lizhennessy** [reply]
Hi there @torressam333! The serverless plan is definitely a great option if you're building a single GraphQL API. This course uses the Enterprise plan because we explore the federation architecture pattern from front to back; running our separate subgraphs and a router in the middle. You don't need to pay for it, however: you can sign up for a [free Enterprise trial here](https://studio.apollographql.com/signup?type=enterprise-trial&referrer=odyssey-ent&_gl=1*zyqh6y*_ga*NjEyOTk0MDguMTcwNjg4NzQxMA..*_ga_0BGG5V2W2K*MTcwNzQ0NzcyMC4xNi4xLjE3MDc0NDc3NTguMC4wLjA.). Do let us know if you have any other questions we can help with!

**Comment by LingVuDev** [comment]
It's such a shame, that I can't do this course without an enterprise account. I really enjoyed your previous introduction course, but then I need to skip this one.

**Reply by lizhennessy** [reply]
Hi @LingVuDev! You can also complete this course with an [Enterprise Trial](https://studio.apollographql.com/signup?type=enterprise-trial&referrer=odyssey-ent&_gl=1*fll486*_ga*NjEyOTk0MDguMTcwNjg4NzQxMA..*_ga_0BGG5V2W2K*MTcwOTIyNTk1My41Ny4xLjE3MDkyMjcwODkuMC4wLjA.)  - it gives you 28 days to test out all of the federation functionality that you'd get through a full-fledged Enterprise account! Hopefully this helps you jump into the world of federation. :)

**Reply by LingVuDev** [reply]
But this not available for private users. I couldn’t create it with my private email.

**Reply by lizhennessy** [reply]
Hey @LingVuDev ! We also have a series all about [GraphOS](https://www.apollographql.com/tutorials/getting-started-with-graphos), which covers many of the same concepts as the Voyage series. This series doesn't require an Enterprise account, but we still walk through creating a supergraph, using `@defer` and other directives, and running schema checks.

**Reply by LingVuDev** [reply]
Thanks @lizhennessy for letting me know. I've bookmarked it.

**Reply by home-grown-engineer** [reply]
You can use DuckDuckGo email forwarding to create a kind of "an enterprise account" :)

**Comment by nsauce** [comment]
When installing via PowerShell using the command -> iwr 'https://rover.apollo.dev/win/latest' | iex 

I found it did not update the binary PATH right away. I had to reboot my computer for it to work and for me to be able to run rover in terminal after.

**Reply by mabuyo** [reply]
Thanks for leaving a note for all PowerShell users!

**Comment by stephan-buckmaster** [comment]
I thought it would show my name when I run `rover config whoami`, Instead I got something like this (userid,api-key details changed):

    $ rover config whoami
    Checking identity of your API key against the registry.
    ┌────────────┬─────────────────────────────────────────────────────────────────────┐
    │ Key Type │ User                                                                  │   
    ├────────────┼─────────────────────────────────────────────────────────────────────┤
    │ User ID  │ fp.1dad39e0-bbb2-4ff2-aa87-8763fcf71dbd                               │   
    ├────────────┼─────────────────────────────────────────────────────────────────────┤
    │ Origin   │ --profile default                                                     │    
    ├────────────┼─────────────────────────────────────────────────────────────────────┤
    │ API Key  │ user***********************************************************xyzA   │
    └────────────┴─────────────────────────────────────────────────────────────────────┘

**Reply by lizhennessy** [reply]
hey @stephan-buckmaster ! That is the correct output from `rover config whoami`. This command is most useful if you have two API keys - each one associated to a different organization in Studio - where you might need to switch between profiles or validate that you're authenticated to the correct org.

**Reply by stephan-buckmaster** [reply]
Thanks @lizhennessy !

**Comment by GershyRbi** [comment]
Mistake in the image under the "Authenticating Rover" section - url is showing github.com, but should probably show studio.apollographql.com

**Reply by mabuyo** [reply]
Great catch, thank you! This will be fixed shortly.

**Comment by aMaverickDragonfruit** [comment]
I followed the README instructions to run the final project on my device. However, I encountered an issue when trying to start the server in the final folder. Specifically, I received a permission denied error when running ./router. Additionally, I was unable to modify the file permissions using chmod. Could you please provide guidance on how to resolve this issue? Thank you!

**Reply by mabuyo** [reply]
Hi there @aMaverickDragonfruit ! It's possible the README might need updating. Did you install the router first before trying to run it? This particular course uses v1.46.0, here's the command to install it: `curl -sSL https://router.apollo.dev/download/nix/v1.46.0 | sh`.
